pkg/model/pipeline.go

```diff
@@ -64,8 +66,48 @@ func BuildCaches(caches map[string]CacheDefinition, existing map[string]Cache) m
 	return existing
 }
 
+func BuildFuncMap(funcs map[string]Function) (tmpl.FuncMap, error) {
+	funcMap := make(tmpl.FuncMap)
+
+	if len(funcs) == 0 {
+		return funcMap, nil
+	}
+
+	// env := Environment{Env: env.NewEnv()}
+	env := NewEnvironment()
+
+	for name, f := range funcs {
+		err := env.Compile(f.Build(name))
+		if err != nil {
+			return nil, fmt.Errorf("cannot compile function: %w", err)
+		}
+
+		ankowrapper := func(args ...interface{}) (interface{}, error) {
+			joined := strings.Trim(strings.Join(strings.Fields(fmt.Sprint(args)), ","), "[]")
+			output, err := env.Execute(fmt.Sprintf(name+"(%s)", joined))
+			if err != nil {
+				return nil, fmt.Errorf("cannot wrap anko function: %w", err)
+			}
+			return output, nil
+		}
+
+		function, err := functions.BuildFunction(f.Params, f.Returns, ankowrapper)
+		if err != nil {
+			return funcMap, errors.New(err.Error() + " for " + name + " function")
+		}
+
+		funcMap[name] = function.Interface()
+	}
+
+	return funcMap, nil
+}
+
 func BuildPipeline(pipeline Pipeline, conf Definition, caches map[string]Cache) (Pipeline, map[string]Cache, error) {
 	caches = BuildCaches(conf.Caches, caches)
+	functions, err := BuildFuncMap(conf.Functions)
+	if err != nil {
+		return nil, nil, errors.New(err.Error())
```

Comment:
On peut retourner l'erreur directement ici

```suggestion
		return nil, nil, err
```